sdk/core/azure-core/CHANGELOG.md

```diff
@@ -8,8 +8,13 @@
 
 ### Bugs Fixed
 
+- Adjusted `AsyncBearerTokenCredentialPolicy` lock initialization logic to use `anyio`.  ([#33307](https://github.com/Azure/azure-sdk-for-python/pull/33307))
```

Comment:
I usually prefer to have changelog targeted for customer consumption, not implementation details. Something like "Adjusted AsyncBearerTokenCredentialPolicy to work properly with trio" which is the actual impact a customer would see.

Reply:
Good point. Updated!